server/src/servershot.js

```diff
@@ -620,42 +620,60 @@ Shot.deleteShot = function(backend, shotId, deviceId, accountId) {
   })
 };
 
-Shot.deleteEverythingForDevice = function(backend, deviceId) {
-  return db.select(
-    `SELECT images.id
-     FROM images JOIN data
-     ON images.shotid = data.id
-     WHERE data.deviceid = $1`,
-    [deviceId]
-  ).then((rows) => {
-      rows.forEach((row) => del(row.id))
-    }
-  ).then(() => {
-    return db.select(
-      `SELECT DISTINCT devices.id
-       FROM devices, devices AS devices2
+Shot.deleteEverythingForDevice = function(backend, deviceId, accountId) {
+  let deviceIdsSelect, deviceIds = [];
+
+  if (accountId) {
+    deviceIdsSelect = db.select(
+      `SELECT devices.id
+       FROM devices
+       WHERE devices.accountid = $1`,
+      [accountId]);
+  } else {
+    deviceIdsSelect = db.select(
```

Comment:
In this case we know exactly what the deviceIds will be: `[deviceId]`. So probably what it should be is something like:

```js
if (accountId) {
  deviceIdsSelect = db.select(...).then((rows) => {
    return rows.map((row) => row.deviceId);
  });
} else {
  deviceIdsSelect = Promise.resolve([deviceId]);
}

Reply:
Just to clarify/confirm: this changes the previous behavior when shots of one device is deleted, if the device is associated to an account (but _not_ authenticated thus no `accountId`), then the shots of the other devices that are associated to that account will also be deleted.  After this patch, only the shots of the given device will be deleted.

Reply:
Really?  That doesn't seem like what I'm reading here. This query seems to get images from all devices:

```js
return db.select(
     `SELECT images.id
      FROM images JOIN data
      ON images.shotid = data.id
      WHERE data.deviceid IN (${db.markersForArgs(1, deviceIds.length)})`,
     deviceIds);
```

Or at least all devices listed in deviceIds...?

Reply:
Yes, you are reading that query correctly.  But if we use `deviceIds = [deviceId]`, then there's only one device id in `IN`.  Currently, the device ids come from https://github.com/mozilla-services/screenshots/blob/master/server/src/servershot.js#L634 regardless of the user's authentication state, and that could returns multiple device ids.

Reply:
If accountId is null then devices.accountid is also null, and there aren't any associated devices. At login or account connect time we set accountid, so we can trust that if it's null then there aren't any other associated deviceIds.